### PR TITLE
Removed FC from kernel driver and use ARM R5 as control program

### PR DESCRIPTION
**Pull request by ZongHong-Lyu**
I am not sure how are we going to maintain support for both C2 and C3 here. C2 has no R5 so can't use this version but C3 forward are going to use R5 and remove FC and LRN. We can discuss in this pull request.

### CODE REVIEW

**Comment by YuyaKobayashi** [comment]
So far this driver does not recognize if the board is C2 or C3, but just read register address for DV700.

It is better to add a device tree entry to show which the board is.

File: src/dmp-dv.c [hunk 1]
```diff
@@ -606,6 +627,10 @@ int drm_register_chrdev(struct drm_dev *drm_dev)
 			}
 
 			rIRQ = drm_dev->subdev[i].irqno;
+			if (i == 0)
+			err = request_irq(rIRQ, handle_int_r5, IRQF_SHARED,
```
Comment (YuyaKobayashi) [inline src/dmp-dv.c]: do indent

**Comment by ZongHong-Lyu** [comment]
Kobayashi-san

I commit a zia-c2 branch for maintain C2 compatibility. The C2 branch is going to be in maintain only mode  so no further development will be on C2.

**Comment by YuyaKobayashi** [comment]
I got it.

**Comment by YuyaKobayashi** [comment]
How about adding kernel parameter for CONV_FREQ to allow users to configure conv_freq permanently?

**Comment by ajkxyz** [comment]
The default frequency can be added as the module parameter for dmp_dv.ko
Or the frequency can be set via startup script like: echo 33 >/sys/class/dmp_dv/dv_conv/conv_freq

And I think it is better to set the default frequency in the boot.bin to the minimum.

**Comment by YuyaKobayashi** [comment]
I agree with using a startup script.

**Comment by ZongHong-Lyu** [comment]
Since the new hardware currently have no svn_version register, I hard coded the version number in driver.
Should we request the hardware to add this register, or we change to use other method to detect hardware features?

**Comment by YuyaKobayashi** [comment]
Yes, I agree that we need some mecanism for HW feature detection.
If we cannot prepare it soon, it is better to update the hardcoded svn_version tentatively.

**Comment by ZongHong-Lyu** [comment]
I think maybe put the hardware capabilities in device tree is one solution. Use some version number to determine feature set seems too limited and hard to maintain.
Else we need the hardware to have some registers of its capabilities.

**Comment by ajkxyz** [comment]
I think it is better to have registers with feature bits

**Comment by ZongHong-Lyu** [comment]
What feature bits are needed?
I checked the user driver and it seems non-squared convolution and deconvolution are checked.
Maybe also dilated? So that's three bits for now.
If this seems ok. I'll ask Ohsawa-san to add one register that reads as 0x7 as the current capabilities bits.

**Comment by ZongHong-Lyu** [comment]
For now I changed the hard coded SVN version to 93, so the check in user driver will work. Ohsawa-san is still not convinced to add caps bits so we need further discussion.

**Pull request merged**